CMakeLists.txt

```diff
@@ -7,6 +7,13 @@ if(NOT CMAKE_BUILD_TYPE)
   set(CMAKE_BUILD_TYPE "RelWithDebInfo" CACHE STRING "Choose build type: Debug Release RelWithDebInfo." FORCE)
 endif()
 
+# LINUX variable is set from 3.25 on
+if (CMAKE_VERSION VERSION_LESS "3.25")
```

Comment:
Are we able to upgrade minimum cmake version to be 3.25 in iOS/Android now?

Reply:
The Android CI is using 3.26.1 and the iOS CI is using 3.25.0. 

What was the reason android or ios blocked a cmake upgrade previously?

Reply:
Updated to 3.25 as minimum but not clear why the macos builds are failing. Based on other successful CI runs the macos build is using cmake 3.26, so the change shouldn't matter. 

https://aiinfra.visualstudio.com/Lotus/_build/results?buildId=306993&view=logs&j=cdb80bd2-8b48-5122-dfc6-e0f4b7485e64&t=793ef336-d55c-5c67-817c-75d71d7f2bbd&l=314
`--     CMake:                       3.26.3`

Error is this:

> 2023-05-11T06:27:58.7961650Z CMake Error at CMakeLists.txt:151 (message):
2023-05-11T06:27:58.7963680Z   CMAKE_FIND_FRAMEWORK is not NEVER

What's the reason CMAKE_FIND_FRAMEWORK must be NEVER? For mac I would have expected it was fine to use the system framework, and cmake defaults to using FIRST in that case: 

https://github.com/Kitware/CMake/blob/13e46189c7f3b39a26e9ca689bc029b7061d26a7/Modules/Platform/Darwin.cmake#L138-L141

```cmake
# default to searching for frameworks first
if(NOT DEFINED CMAKE_FIND_FRAMEWORK)
  set(CMAKE_FIND_FRAMEWORK FIRST)
endif()
```

However the CMake default hasn't changed any time recently. Same in 3.25.0 as 3.26.4 so how a build that appears to be using CMake 3.26 fails when the minimum cmake version changes to 3.25 is a bit of a mystery.

Reply:
One consideration is to be equal of less than ORT cmake version requirements due to combined source build.
CMAKE_FIND_FRAMEWORK can be updated if CI pipeline passes.

Reply:
> One consideration is that cmake version here should be equal of less than ORT cmake version requirements due to combined source build.

You are free to choose your requirements. If ORT requires cmake version >=A  and ORT-ext requires cmake version >=B, the combined build will require cmake version >= max(A,B). That's it.

Reply:
ORT is using 3.25 or higher for everything except ROCm.

Re: build failure when minimum version is 3.25, I did some builds with this extra logging

```cmake
if (CMAKE_FIND_FRAMEWORK)
  message(STATUS "CMAKE_FIND_FRAMEWORK is ${CMAKE_FIND_FRAMEWORK} prior to us calling set. CMAKE_VERSION=${CMAKE_VERSION}")
endif()

set(CMAKE_FIND_FRAMEWORK NEVER CACHE STRING "...")

if(NOT "${CMAKE_FIND_FRAMEWORK}" STREQUAL "NEVER")
  message(FATAL_ERROR "CMAKE_FIND_FRAMEWORK is ${CMAKE_FIND_FRAMEWORK} not NEVER.")
endif()
```

With a build with min version of 3.20, CMAKE_FIND_FRAMEWORK has a default value and the `set` updates it as the only output is:

`-- CMAKE_FIND_FRAMEWORK is FIRST prior to us calling set. CMAKE_VERSION=3.26.3`

With a build with min version of 3.25, it also has a default value but the `set` doesn't update the value. Both builds have an actual cmake version of 3.26.3

```
-- CMAKE_FIND_FRAMEWORK is FIRST prior to us calling set.  CMAKE_VERSION=3.26.3
CMake Error at CMakeLists.txt:155 (message):
  CMAKE_FIND_FRAMEWORK is FIRST not NEVER.
```

Looks like it's due to a [policy](https://cmake.org/cmake/help/latest/policy/CMP0126.htm) that was added in 3.21 which appears to be applied based on the minimum cmake version rather than actual. Builds fine if I add ` cmake_policy(SET CMP0126 OLD)` however that doesn't seem like a good approach going forward given it's using the old version of the policy and that is explicitly deprecated. I tried adding `FORCE` to the `set` didn't help. Setting regular and cache variables does work.